test/SlowTests/Issues/RavenDB-16782.cs

```diff
@@ -0,0 +1,47 @@
+﻿using System.Linq;
+using System.Threading.Tasks;
+using FastTests;
+using Raven.Client.Documents;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace SlowTests.Issues
+{
+    public class RavenDB_16782 : RavenTestBase
+    {
+        public RavenDB_16782(ITestOutputHelper output) : base(output)
+        {
+        }
+
+        public async Task ShouldWork()
+        {
+            using (var store = GetDocumentStore())
+            {
+                var dateTime = "2015-10-17T13:28:17-05:00";
+                using (var session = store.OpenAsyncSession())
+                {
+                    await session.StoreAsync(new User
+                    {
+                        DateTime = dateTime
+                    });
+
+                    await session.SaveChangesAsync();
+                }
+
+                using (var session = store.OpenAsyncSession())
+                {
+                    var result = await session.Query<User>().Where(x => x.DateTime == dateTime).ToListAsync();
+                    Assert.Equal(1, result.Count);
+
+                    result = await session.Query<User>().Where(x => x.DateTime == "2015-10-17T13:28:17-04:00").ToListAsync();
+                    Assert.Equal(0, result.Count);
+                }
+            }
+        }
+
+        private class User
+        {
+            public string DateTime { get; set; }
```

Comment:
Maybe I'm missing something here, but shouldn't this be DateTime type?
And if it is supposed to be a string type, how is this test related to DateTime? All I see are string comparisons, no?

Reply:
We are skipping here Client's internal behavior where DateTimeOffset is sent using `o` format instead of `ISO-8601` for which we are adding support in this PR.